dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -626,6 +627,97 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
 
     TypedDataset.create[Out](selected)
   }
+
+  /**
+    *
+    * @see [[frameless.TypedDataset.withColumnApply#apply]]
+    * @tparam U the output type
+    */
+  def withColumnTyped[U] = new withColumnApply[U]
+
+  class withColumnApply[U] {
+    /**
+      *  Adds a column to a Dataset so long as the specified output type, `U`, has
```

Comment:
I would move this documentation to the method and leave the class undocumented, supposingly users are never going to see this type in IDE autocomplete so that should be fine.